cpp/autosar/src/rules/A5-1-1/LiteralValueUsedOutsideTypeInit.ql

```diff
@@ -26,6 +45,11 @@ where
   not exists(ConstructorCall cc | cc.getAnArgument() = l) and
   not exists(ConstructorFieldInit cf | cf.getExpr() = l) and
   not l = any(LoggingOperation logOp).getALoggedExpr().getAChild*() and
+  not l = any(FileStreamFunctionCall fsc).getAnArgument().getAChild*() and
```

Comment:
it is not, which is why we were seeing some of those FPs,

this is what I have so far but advice would be helpful, there is a trade off:
```
class FileStreamLogging extends LoggingOperation, FileStreamFunctionCall {
  override Expr getALoggedExpr() { result = getAnArgument() }

  override Expr getFStream() {
    result = this.getQualifier()
  }
}
```
it feels a bit weird for the extension point for this, since both base classes we are trying to represent here are abstract... I mean its possible to write but is it strange?

additionally, 
it would deprecate the need for the 
```
not exists(FunctionCall fc, LoggerOrStreamWrapperFunction w |
    fc.getAnArgument() = l and w.getACallToThisFunction() = fc
   ) and
```
this part of the query because I think the `getALoggedExpr()` predicate returns the wrapper function calls,

but that does introduce a false negative that we werent getting before on a case like this:
```
void test_stream_two(std::ostream &os, const char *str,
                     const char *alt) noexcept {
  os << str << "logging string"; // COMPLIANT - literal used in stream write
  throw alt;
}

void test_not_wrapper_stream(std::ostream &os, const char *str) noexcept {
  test_stream_two(os, "test", "not okay"); // NON_COMPLIANT - test_stream_two is
                                           // not actually exclusively a wrapper (but no longer detected if we use the above FileStreamLogging class
  
}
```

let me know which seems more sensible to go with! thanks!

Reply:
decided to just go with it as presented above